app/models/application_experience.rb

```diff
@@ -1,3 +1,16 @@
 class ApplicationExperience < ApplicationRecord
+  belongs_to :application_form, touch: true
+  belongs_to :experienceable, polymorphic: true, optional: true
+
+  after_save -> { update_experienceable }, if: -> { experienceable.nil? }
+
   validates :role, :organisation, :start_date, presence: true
+
+private
+
+  def update_experienceable
+    without_auditing do
+      update!(experienceable: application_form)
```

Comment:
Have we considered how we could do this without an additional database transaction? I'm assuming this `.update!` will happen before the previous action has been committed 🤔
Alternatively we could set the `experienceable` before it gets saved - we could possibly do this without an `update` but be setting the value ie. `self.experienceable = application_form` or `self.experienceable_id = application_form_id` and `self.experienceable_type = "ApplicationForm"`